src/thruster.cpp

```diff
@@ -107,8 +104,16 @@ void Thruster::ReloadParams()
 {
     if(!ros::param::getCached("visualize_thrusters", visualize_thrusters))
     {
-        ROS_WARN_STREAM("failed to load thruster visualizer on/off");
+        ROS_WARN("failed to load thruster visualizer on/off");
+    }
+
+    double visualizer_update_rate = 0.1;
+    if(!ros::param::getCached("visualizer_update_rate",
+                visualizer_update_rate))
+    {
+        ROS_WARN("no visualizer update rate specified");
     }
+    visualizer_update_time = ros::Duration(1.0/visualizer_update_rate);
```

Comment:
If there's no default value the duration will end up being 0 and the loop will be run every frame (which slows gazebo to a near-halt)